crypto/s2n_hkdf.c

```diff
@@ -213,6 +212,10 @@ int s2n_hkdf_extract(struct s2n_hmac_state *hmac, s2n_hmac_algorithm alg, const
     POSIX_ENSURE_REF(key);
     POSIX_ENSURE_REF(pseudo_rand_key);
 
+    uint8_t hmac_size = 0;
+    POSIX_GUARD(s2n_hmac_digest_size(alg, &hmac_size));
+    POSIX_ENSURE(hmac_size <= pseudo_rand_key->size, S2N_ERR_HKDF_OUTPUT_SIZE);
+
```

Comment:
You have to repeat this check before modifying the blob in s2n_custom_hkdf_extract anyway, so I think it would actually be clearer if you just duplicated the check in s2n_libcrypto_hkdf_expand. It's not much code.

Reply:
Or I suppose you could set pseudo_rand_key->size here instead of in s2n_custom_hkdf_extract, but I'm not sure how s2n_libcrypto_hkdf_expand would feel about that.

Reply:
I added a new comment to the libcrypto extract, and I would rather the comment where the check is. So I think it makes sense for the checks to be separate for that reason also.